### PR TITLE
Bugfix: Intel oneAPI 2022.2 Fortran

### PR DESCRIPTION
**Pull request by gardner48**
Fix compilation error when building the serial NVector Fortran 2003 example with the Intel oneAPI 2022.2 Fortran compiler.

### CODE REVIEW

File: doc/kinsol/guide/source/Introduction.rst [hunk 1]
```diff
@@ -93,6 +93,9 @@ Changes in v6.4.1
 
 Fixed a bug with the Kokkos interfaces that would arise when using clang.
 
+Fixed a compilation error with the Intel oneAPI 2022.2 Fortran compiler in the
+Fortran 2003 interface test for the serial NVector.
```
Comment (balos1) [inline doc/kinsol/guide/source/Introduction.rst:97]:
```
Fortran 2003 interface test for the serial ``N_Vector``.
```

File: doc/cvode/guide/source/Introduction.rst [hunk 2]
```diff
@@ -116,6 +116,9 @@ Changes in v6.4.1
 
 Fixed a bug with the Kokkos interfaces that would arise when using clang.
 
+Fixed a compilation error with the Intel oneAPI 2022.2 Fortran compiler in the
+Fortran 2003 interface test for the serial NVector.
```
Comment (balos1) [inline doc/cvode/guide/source/Introduction.rst:120]:
```suggestion
Fortran 2003 interface test for the serial ``N_Vector``.
```

File: doc/arkode/guide/source/Introduction.rst [hunk 3]
```diff
@@ -123,6 +123,9 @@ Changes in v5.4.1
 
 Fixed a bug with the Kokkos interfaces that would arise when using clang.
 
+Fixed a compilation error with the Intel oneAPI 2022.2 Fortran compiler in the
+Fortran 2003 interface test for the serial NVector.
```
Comment (balos1) [inline doc/arkode/guide/source/Introduction.rst:127]:
```suggestion
Fortran 2003 interface test for the serial ``N_Vector``.
```

File: doc/ida/guide/source/Introduction.rst [hunk 4]
```diff
@@ -77,6 +77,9 @@ Changes in v6.4.1
 
 Fixed a bug with the Kokkos interfaces that would arise when using clang.
 
+Fixed a compilation error with the Intel oneAPI 2022.2 Fortran compiler in the
+Fortran 2003 interface test for the serial NVector.
```
Comment (balos1) [inline doc/ida/guide/source/Introduction.rst:81]:
```suggestion
Fortran 2003 interface test for the serial ``N_Vector``.
```

File: doc/kinsol/guide/source/Introduction.rst (hunk 1 above)
Comment (balos1) [inline doc/kinsol/guide/source/Introduction.rst:97]:
```suggestion
Fortran 2003 interface test for the serial ``N_Vector``.
```

File: doc/idas/guide/source/Introduction.rst [hunk 5]
```diff
@@ -91,6 +91,9 @@ Changes in v5.4.1
 
 Fixed a bug with the Kokkos interfaces that would arise when using clang.
 
+Fixed a compilation error with the Intel oneAPI 2022.2 Fortran compiler in the
+Fortran 2003 interface test for the serial NVector.
```
Comment (balos1) [inline doc/idas/guide/source/Introduction.rst:95]:
```suggestion
Fortran 2003 interface test for the serial ``N_Vector``.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,9 @@
 
 Fixed a bug with the Kokkos interfaces that would arise when using clang.
 
+Fixed a compilation error with the Intel oneAPI 2022.2 Fortran compiler in the
+Fortran 2003 interface test for the serial NVector.
```
Comment (balos1) [inline CHANGELOG.md:8]:
```suggestion
Fortran 2003 interface test for the serial `N_Vector`.
```

**Review by balos1 — CHANGES_REQUESTED** [review]
Minor suggestion for the change log, otherwise this looks good.

File: doc/cvodes/guide/source/Introduction.rst [hunk 7]
```diff
@@ -116,6 +116,9 @@ Changes in v6.4.1
 
 Fixed a bug with the Kokkos interfaces that would arise when using clang.
 
+Fixed a compilation error with the Intel oneAPI 2022.2 Fortran compiler in the
+Fortran 2003 interface test for the serial NVector.
```
Comment (gardner48) [inline doc/cvodes/guide/source/Introduction.rst:120]:
```suggestion
Fortran 2003 interface test for the serial ``NVector``.
```

**Review by balos1 — APPROVED** [review]

**Review by drreynolds — APPROVED** [review]

**Pull request merged**